src/ansible_compat/prerun.py

```diff
@@ -361,8 +364,11 @@ def ansible_config_get(key: str, kind: Type[Any] = str) -> Union[str, List[str],
                 raise RuntimeError(f"Unexpected data read for {key}: {val}")
             return val
     else:
-        raise RuntimeError("Unknown data type.")
-    return None
+        raise NotImplementedError("Unknown data type %s." % kind)
+
+    if default == SENTINEL:
```

Comment:
@ssbarnea the idiom is the same as with `None` — you must check for the identity, not value:
```suggestion
    if default is SENTINEL:
```